addons/web/static/tests/fields/relational_fields/field_many2many_tests.js

```diff
@@ -828,6 +828,109 @@ QUnit.module('fields', {}, function () {
             form.destroy();
         });
 
+        QUnit.test('many2many field with link/unlink options (list)', async function (assert) {
+            assert.expect(5);
+
+            this.data.partner.records[0].timmy = [12, 14];
+
+            const form = await createView({
+                View: FormView,
+                model: 'partner',
+                data: this.data,
+                arch: `
+                    <form>
+                        <field name="color"/>
+                        <field name="timmy" options="{'link': [('color', '=', 'red')], 'unlink': [('color', '=', 'red')]}">
+                            <tree>
+                                <field name="name"/>
+                            </tree>
+                        </field>
+                    </form>`,
+                archs: {
+                    'partner_type,false,list': '<tree><field name="name"/></tree>',
+                    'partner_type,false,search': '<search/>',
+                },
+                res_id: 1,
+                viewOptions: {
+                    mode: 'edit',
+                },
+            });
+
+            // color is red -> link and unlink actions are available
+            assert.containsOnce(form, '.o_field_x2many_list_row_add',
+                "should have the 'Add an item' link");
+            assert.containsN(form, '.o_list_record_remove', 2,
+                "should have two remove icons");
+
+            await testUtils.dom.click(form.$('.o_field_x2many_list_row_add a'));
+
+            assert.containsN(document.body, '.modal .modal-footer button', 3,
```

Comment:
Add a comment (or a test) to make explicit that the three buttons are: Select, Create Discard, and, most importantly, that the unlink/link feature only has impact on the Select Button. It should be explicit that a case where only select/discard is possible.
This begs the question: should a case where only create/discard be possible; i.e. only creation of a record without linking, within a m2m ?
In my opinion, it should not, a m2m field should not be the place to create a record for another model.....